packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -243,10 +266,13 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
             return;
         }
 
-        const newText = await this.readContents();
-        if (newText === undefined || token.isCancellationRequested || this._dirty) {
+        let newText = await this.readContents();
+        if (token.isCancellationRequested || this._dirty) {
             return;
         }
+        if (newText === undefined) {
```

Comment:
You create a monaco model with some text and then save, after that a file gets deleted on fs, monaco model is not ditry, but still in memory, so it gets reset to empty content with `valid` flag false.

Reply:
Actually maybe we don't need to reset content to empty string anymore, i did it before introducing `valid` flag, but since we can distinguish now there model is really loaded from resource it can be unnecessary. I will try to revert it.